pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -118,9 +120,27 @@ testing::AssertionResult PilzJointTrajectoryControllerTest::isControllerInUnhold
   return testing::AssertionSuccess();
 }
 
-///////////////////////////////////////
-//    The actual tests start here    //
-///////////////////////////////////////
+//////////////////////////////////
+//  Testing of Initialization   //
+//////////////////////////////////
+
+/**
+ * @brief Test if test environemt is initialized successfully.
+ */
+TEST_F(PilzJointTrajectoryControllerTest, testInitializiation)
+{
+  ASSERT_TRUE(manager_->loadController()) << "Failed to initialize the controller.";
+
+  EXPECT_TRUE(ros::service::exists(CONTROLLER_NAMESPACE + HOLD_SERVICE, true));
+  EXPECT_TRUE(ros::service::exists(CONTROLLER_NAMESPACE + UNHOLD_SERVICE, true));
+  EXPECT_TRUE(ros::service::exists(CONTROLLER_NAMESPACE + IS_EXECUTING_SERVICE, true));
+
+  EXPECT_TRUE(action_client_.waitForActionServer());
+}
+
+/////////////////////////////////////////
+//    Testing of additional methods    //
+/////////////////////////////////////////
```

Comment:
If we have a test file consisting of so many tests, that we need comments to visually "split" the file, maybe it is time to actually split the tests into separate test files. Currently the file consists of 654 lines...

Reply:
I introduced 2 new files in order to reduced the tests per file.